jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/dao/MySqlDatabaseAccessor.java

```diff
@@ -26,7 +26,7 @@ protected String addLimitAndOffsetToQuery(String sql, int limit, int offset) {
       return addLimitToQuery(sql, limit);
     }
     else {
-      return sql + " LIMIT " + limit + "," + offset;
+      return sql + " LIMIT " + offset + "," + limit;
```

Comment:
Actually nm. Will do in follow up also. I've tested the current version on real cluster with MySQL. Will redo at a later point.